### ISSUE TITLE
Unhandled exceptions when resources in state don't exist.

### ISSUE THREAD

**Issue by jomadu** [JupiterOne/terraform-provider-jupiterone#211]
### Terraform Version

```
Terraform v1.6.6
on darwin_amd64
+ provider registry.terraform.io/jupiterone/jupiterone v1.4.3
```

### Affected Resource(s)

- jupiterone_dashboard - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_dashboard.go#L136
- jupiterone_framework - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_framework.go#L223
- jupiterone_frameworkitem - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_frameworkitem.go#L185
- jupiterone_group - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_group.go#L177
- jupiterone_libraryitem - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_libraryitem.go#L177
- jupiterone_rule - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_rule.go#L447
- jupiterone_user_group - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_user_group.go#L174
- jupiterone_user_group_membership - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_user_group_membership.go#L182
- jupiterone_widget - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_widget.go#L151

### Unaffected Resource(s)

- jupiterone_question - https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/jupiterone/resource_question.go#L267

### Misleading Documentation

[README](https://github.com/JupiterOne/terraform-provider-jupiterone/blob/main/README.md?plain=1#L220) documentation recommends this poor pattern for handling errors.

It should instead recommend removing the resource from state, so that it can be recreated.

For example,

```go
func (r *J1EntityResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
	var data J1EntityModel

	// Read Terraform state into the model
	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)

	if resp.Diagnostics.HasError() {
		return
	}

        // Grab your entity from jupiterone gql api and then map it to the J1EntityModel
        // See other resource.go files for examples
	entity, err := client.GetJ1Entity(ctx, r.qlient, data.Id.ValueString())
        // If the error is a not found error, we should remove the resource so it can be recreated
	if err != nil {
                if strings.Contains(err.Error(), "does not exist") {
			resp.State.RemoveResource(ctx)
			return
		} else {
		        resp.Diagnostics.AddError("failed to get j1 entity", err.Error())
		        return
                }
	}

	data.Name = types.StringValue(entity.IamGetGroup.GroupName)
	data.Description = types.StringValue(entity.IamGetGroup.GroupDescription)

	// Save updated data into Terraform state
	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
}
```

### Terraform Configuration Files

This example is for a user group resource, but the same bug applies to all the listed resources affected above.

```
resource "jupiterone_user_group" "admins" {
  name        = "admins-managed"
  description = "Admins"
  permissions = ["accessAdmin"]
}
```

### Debug Output

Link to gist of debug output: https://gist.github.com/jomadu/fdebc36286e90005aa97f83d7a61c68e

### Panic Output

N/A

### Expected Behavior

This example is for a user group resource, but the same bug applies to all the listed resources affected above.

1. A user group is created by terraform during a terraform apply
2. The user group is deleted by a user
3. Running terraform plan should recognize that the user group was deleted, and needs to be re-created

### Actual Behavior

This example is for a user group resource, but the same bug applies to all the listed resources affected above.

1. A user is created by terraform during a terraform apply
2. The user group is deleted by a user
3. Running terraform plan results in the unhandled error `Error: failed to get user group`

### Steps to Reproduce

This example is for a user group resource, but the same bug applies to all the listed resources affected above.

1. Create a jupiterone_user_group resource using terraform apply
2. Delete the user group using a direct api call, or deleting the user group through the JupiterOne UI
3. Run terraform plan
4. Observe in the unhandled error `Error: failed to get user group`

### References

- https://github.com/JupiterOne/terraform-provider-jupiterone/issues/202

**Comment by hesto2** [comment]
Thanks for the report, we'll get this in the coming sprint!

**Comment by hesto2** [comment]
Hey there, this should be fixed in `v1.5.0`
